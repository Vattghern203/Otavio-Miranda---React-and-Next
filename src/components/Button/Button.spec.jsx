import { fireEvent, render, screen } from "@testing-library/react"
import { Button } from '.'
import userEvent from "@testing-library/user-event";

describe('<Button />', () => {
    it('should render the button with the text "load more"', () => {
        render(<Button text="Load More" />);

        expect.assertions(1)

        const button = screen.getByRole('button', { name: /load more/i })
        expect(button).toBeInTheDocument()  
    })

    it('should call a function on button click', () => {

        const fn = jest.fn()

        render(<Button text="Load More" onClick={fn} />)

        const button = screen.getByRole('button', { name: /load more/i})

        //fireEvent.click(button)
        userEvent.click(button)

        expect(fn).toHaveBeenCalled()
    })

    it('should be disabled when disable is equals to true', () => {
        render(<Button text="Load More" disabled={true} />)

        const button = screen.getByRole('button', { name: /load more/i})
        
        expect(button).toBeDisabled()
    })

    it('should be enabled when disable is equals to false', () => {
        render(<Button text="Load More" disabled={false} />)

        const button = screen.getByRole('button', { name: /load more/i})
        
        expect(button).toBeEnabled()
    })

    it('should match the snapshot', () => {
        const { container } = render(<Button text="load more" disabled={false} />)
        // eslint-disable-next-line testing-library/no-node-access
        expect(container.firstChild).toMatchSnapshot()
    })
})