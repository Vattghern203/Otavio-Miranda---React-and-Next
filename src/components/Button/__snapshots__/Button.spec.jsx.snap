// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
<button
  class="button"
>
  load more
</button>
`;
